### ISSUE TITLE
Long startup time (Geoserver 2.8.x)

### ISSUE THREAD

**Issue by danfruehauf** [aodn/geoserver-build#34]
It's something I've observed on my VM, connecting to production DB. Geoserver takes minutes to start. After loading all features, stores and layers, it sort of hangs and run a lot of geometry queries on the DB, such as:
```
SELECT SRID FROM GEOGRAPHY_COLUMNS WHERE F_TABLE_SCHEMA = 'srs_altimetry' AND F_TABLE_NAME = 'srs_altimetry_timeseries_data' AND F_GEOGRAPHY_COLUMN = 'geom'
```

Before it hangs, those are the log lines:
```
16 Oct 09:11:36 INFO [gwc.layer] - GeoServer TileLayer store base directory is: /vagrant/src/geoserver/gwc-layers
16 Oct 09:11:36 INFO [gwc.layer] - Loading tile layers from /vagrant/src/geoserver/gwc-layers
16 Oct 09:11:36 INFO [gwc.layer] - GWC configuration based on GeoServer's Catalog loaded successfuly
16 Oct 09:11:36 INFO [layer.TileLayerDispatcher] - Configuration GeoServer Catalog Configuration contained no layers.
16 Oct 09:11:36 INFO [config.XMLConfiguration] - Initializing GridSets from /vagrant/src/geoserver/gwc
16 Oct 09:11:36 INFO [config.XMLConfiguration] - Initializing layers from /vagrant/src/geoserver/gwc
16 Oct 09:11:36 INFO [layer.TileLayerDispatcher] - Configuration /vagrant/src/geoserver/gwc contained no layers.
16 Oct 09:11:36 INFO [gwc.config] - Initializing GeoServer specific GWC configuration from gwc-gs.xml
```

Eventually, after a few minutes, it starts and spits the following log lines:
```
16 Oct 09:14:19 WARN [geotools.jdbc] - Could not find mapping for 'duration', ignoring the column and setting the feature type read only
```

And once it's actually up:
```
16 Oct 09:16:43 INFO [diskquota.ConfigLoader] - DiskQuota configuration not found: /vagrant/src/geoserver/gwc/geowebcache-diskquota.xml
16 Oct 09:16:43 INFO [diskquota.ConfigLoader] - DiskQuota configuration not found: /vagrant/src/geoserver/gwc/geowebcache-diskquota.xml
16 Oct 09:16:43 INFO [diskquota.DiskQuotaMonitor] - Setting up disk quota periodic enforcement task
16 Oct 09:16:43 INFO [diskquota.DiskQuotaMonitor] - 0 layers configured with their own quotas. 
16 Oct 09:16:43 INFO [diskquota.DiskQuotaMonitor] - 0 layers attached to global quota 500.0 MB
16 Oct 09:16:43 INFO [diskquota.DiskQuotaMonitor] - Disk quota periodic enforcement task set up every 10 SECONDS
16 Oct 09:16:43 INFO [ows.OWSHandlerMapping] - Mapped URL path [/gwc/service/**] onto handler 'dispatcher'
16 Oct 09:16:43 INFO [geowebcache.GeoWebCacheDispatcher] - Invoked setServletPrefix(gwc)
16 Oct 09:16:43 INFO [georss.GeoRSSPoller] - Initializing GeoRSS poller in a background job...
16 Oct 09:16:43 INFO [georss.GeoRSSPoller] - No enabled GeoRSS feeds found, poller will not run.
16 Oct 09:16:43 INFO [rest.RESTDispatcher] - Created RESTDispatcher with 15 paths
16 Oct 09:16:43 INFO [wms.WMSService] - Will NOT recombine tiles for non-tiling clients.
16 Oct 09:16:43 INFO [wms.WMSService] - Will proxy requests to backend that are not getmap or getcapabilities.
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/kml] onto handler 'dispatcher'
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/kml/*] onto handler 'dispatcher'
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/kml/icon/**/*] onto handler 'kmlIconService'
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/ows/**] onto handler 'dispatcher'
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/ows] onto handler 'dispatcher'
16 Oct 09:16:44 INFO [org.geoserver] - GeoServer configuration lock is enabled
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wcs] onto handler 'dispatcher'
16 Oct 09:16:44 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wcs/**] onto handler 'dispatcher'
16 Oct 09:16:46 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wfs/*] onto handler 'dispatcher'
16 Oct 09:16:46 INFO [ows.OWSHandlerMapping] - Mapped URL path [/TestWfsPost] onto handler 'wfsTestServlet'
16 Oct 09:16:46 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wfs] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wms/*] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wms] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/animate] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/animate/*] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wps] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/wps/*] onto handler 'dispatcher'
16 Oct 09:16:53 INFO [ows.OWSHandlerMapping] - Mapped URL path [/temp/**] onto handler 'filePublisher'
16 Oct 09:16:53 INFO [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input zones of type class java.lang.Object cannot be handled
16 Oct 09:16:53 INFO [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input nodata of type class it.geosolutions.jaiext.range.Range cannot be handled
16 Oct 09:16:53 INFO [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input rangeData of type class java.lang.Object cannot be handled
16 Oct 09:16:53 INFO [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the output zonal statistics of type interface java.util.List cannot be handled
16 Oct 09:16:53 INFO [geoserver.wps] - Found 14 bindable processes in Raster processes
16 Oct 09:16:53 INFO [geoserver.wps] - Found 48 bindable processes in Geometry processes
16 Oct 09:16:53 INFO [geoserver.wps] - Found 7 bindable processes in GeoServer specific processes
16 Oct 09:16:53 INFO [geoserver.wps] - Found 29 bindable processes in Vector processes
16 Oct 09:16:54 INFO [geoserver.wps] - Found 89 bindable processes in Deprecated processes
16 Oct 09:16:54 INFO [wps.NetcdfOutputProcess] - constructor
16 Oct 09:16:54 INFO [geoserver.security] - Start reloading user/groups for service named default
16 Oct 09:16:54 INFO [geoserver.security] - Reloading user/groups successful for service named default
16 Oct 09:16:54 INFO [geoserver.security] - AuthenticationCache Initialized with 1000 Max Entries, 300 seconds idle time, 600 seconds time to live and 3 concurrency level
16 Oct 09:16:54 INFO [geoserver.security] - AuthenticationCache Eviction Task created to run every 600 seconds
```

More information:
 * Tested inside po box
 * Geoserver doesn't start because it doesn't respond to any HTTP requests during this time.
 * Production DB means 2-aws-syd, so no NSP awkwardness involved

When geoserver finally starts it works alright.

**Comment by jonescc** [comment]
@danfruehauf - looks to me like the long startup time is due to geowebcache initialising.

We don't use geowebcache?  Perhaps just need to disable it.

**Comment by danfruehauf** [comment]
@jonescc Yes, makes sense. Is it actually solving the problem?

**Comment by jonescc** [comment]
Not sure how to turn it off.  Disabled all the caching options and it  still runs on startup.  Suggestions on the web are to remove gwc*.jar from lib directory - but then of course falls over trying to load the beans.

Need to shelve this for the moment.

**Comment by jonescc** [comment]
Actually found another couple of gs gwc jars - deleted them and geowebcache config no longer happens - and yes startup time is much faster

**Comment by danfruehauf** [comment]
We should incorporate it in the build then.

**Comment by jonescc** [comment]
Fixed by ce69d0a4f4cbd6ce70715505b7c81a4488437744
